internal/http/pprof.go

```diff
@@ -0,0 +1,24 @@
+package http
+
+import (
+	"net/http"
+	"net/http/pprof"
+)
+
+// RegisterPprof registers all pprof handlers and the index on the provided ServeMux.
+func RegisterPprof(mux *http.ServeMux) {
```

Comment:
nittiest of nits/question: did you want to make this `RegisterPProf` to match `rpc.PProf`, or do we intentionally follow different naming conventions for http vs rpc?